stories/modules-tagmanager-setup.stories.js

```diff
@@ -0,0 +1,167 @@
+/**
+ * Tag Manager Module Setup Stories.
+ *
+ * Site Kit by Google, Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import { storiesOf } from '@storybook/react';
+
+/**
+ * WordPress dependencies
+ */
+import { removeAllFilters, addFilter } from '@wordpress/hooks';
+
+/**
+ * Internal dependencies
+ */
+import { WithTestRegistry, createTestRegistry, freezeFetch } from '../tests/js/utils';
+import { fillFilterWithComponent } from '../assets/js/util';
+import SetupWrapper from '../assets/js/components/setup/setup-wrapper';
+import { STORE_NAME as CORE_SITE, AMP_MODE_PRIMARY, AMP_MODE_SECONDARY } from '../assets/js/googlesitekit/datastore/site/constants';
+import { STORE_NAME as CORE_USER } from '../assets/js/googlesitekit/datastore/user';
+import { SetupMain as TagManagerSetup } from '../assets/js/modules/tagmanager/components/setup';
+import { STORE_NAME, ACCOUNT_CREATE } from '../assets/js/modules/tagmanager/datastore/constants';
+import * as fixtures from '../assets/js/modules/tagmanager/datastore/__fixtures__';
+
+function Setup( props ) {
+	global._googlesitekitLegacyData.setup.moduleToSetup = 'tagmanager';
+
+	removeAllFilters( 'googlesitekit.ModuleSetup-tagmanager' );
+	addFilter(
+		'googlesitekit.ModuleSetup-tagmanager',
+		'googlesitekit.TagManagerModuleSetup',
+		fillFilterWithComponent( TagManagerSetup )
+	);
+
+	return (
+		<WithTestRegistry { ...props }>
+			<SetupWrapper />
+		</WithTestRegistry>
+	);
+}
+
+storiesOf( 'Tag Manager Module/Setup', module )
+	.addDecorator( ( storyFn ) => {
+		const registry = createTestRegistry();
+		registry.dispatch( STORE_NAME ).setSettings( {} );
+		registry.dispatch( STORE_NAME ).receiveGetExistingTag( null );
+		registry.dispatch( CORE_USER ).receiveGetAuthentication( {} );
+
+		return storyFn( registry );
+	} )
```

Comment:
Same for setup, we could at some point update other Storybook files to use a decorator and expanded `Setup` component like here.